core/modules/parser/QSMySqlListener.cc

```diff
@@ -852,13 +873,13 @@ class BinaryComparasionPredicateAdapter :
         } else if (_right == nullptr) {
             _right = valueExpr;
         } else {
-            CHECK_EXECUTION_CONDITION(false, "left and right values must be set only once.")
+            CHECK_EXECUTION_CONDITION(false, "left and right values must be set only once.", _ctx)
         }
     }
 
     void onExit() {
         CHECK_EXECUTION_CONDITION(_left != nullptr && _right != nullptr,
```

Comment:
Perhaps just: _left && _right ? The shared_ptr already has this operator:
http://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool

Reply:
yeah; my preference when given the option is usually the verbose route.